File: app/src/main/java/tech/pauly/findapet/data/models/Photo.java
package tech.pauly.findapet.data.models;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.Text;

@Root(name = "photo")
public class Photo {

    @Attribute
    private String id;

    @Attribute
    private String size;

    @Text
    private String url;

    public String getId() {
        return id;
    }

    public PhotoSize getSize() {
        return PhotoSize.Companion.fromString(size);
    }

    public String getUrl() {
        return url;
    }
}

File: app-old/src/main/java/tech/pauly/old/data/models/Photo.kt
package tech.pauly.old.data.models

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Root
import org.simpleframework.xml.Text

@Root(name = "photo", strict = false)
open class Photo {

    @field:Attribute(name = "id")
    open lateinit var id: String
        protected set

    @field:Attribute(name = "size")
    private lateinit var _size: String
    open val size: PhotoSize
        get() = PhotoSize.fromString(_size)

    @field:Text
    open lateinit var url: String
        protected set
}
